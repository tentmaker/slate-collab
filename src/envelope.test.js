import { Value } from 'slate';
import { init } from 'automerge';

import Envelope from './envelope';

const value = Value.fromJSON({
  document: {
    nodes: [
      {
        object: 'block',
        type: 'line',
        nodes: [
          {
            object: 'text',
            leaves: [
              {
                object: 'leaf',
                text: 'foobar',
              },
            ],
          },
        ],
      },
    ],
  },
});

it('should initialize with a slate value', () => {
  const envelope = new Envelope(value);
  expect(envelope.value().document.text).toBe('foobar');
});

it('should initialize with a serialized automerge document', () => {
  const serialized = new Envelope(value).serialize();
  const envelope = new Envelope(serialized);
  expect(envelope.value().document.text).toBe('foobar');
});

it('should throw an error if no initializer is provided', () => {
  expect(() => {
    const envelope = new Envelope();
  }).toThrowErrorMatchingSnapshot();
});
