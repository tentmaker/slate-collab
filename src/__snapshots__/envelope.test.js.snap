// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should throw an error if no initializer is provided 1`] = `"An Envelope must be initialized with either a serialized Envelope string or a Slate Value."`;
